spring-integration-cloudevents/src/test/java/org/springframework/integration/cloudevents/transformer/ToCloudEventTransformerTests.java

```diff
@@ -0,0 +1,330 @@
+		Message<byte[]> message = createBaseMessage(payload, eventFormat.serializedContentType())
+				.setHeader("customheader", "test-value")
+				.setHeader("otherheader", "other-value")
+				.build();
+		Message<byte[]> result = transformMessage(message, this.transformerWithNoExtensions);
+		return eventFormat.deserialize(result.getPayload());
+	}
+
+	@SuppressWarnings("unchecked")
+	private Message<byte[]> transformMessage(Message<byte[]> message, ToCloudEventTransformer transformer) {
+		Object result = transformer.doTransform(message);
+
+		assertThat(result).isNotNull();
+		assertThat(result).isInstanceOf(Message.class);
+		return (Message<byte[]>) result;
+	}
+
+	private byte[] convertPayloadToBytes(TestRecord testRecord) throws Exception {
+		ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+		ObjectOutputStream out = new ObjectOutputStream(byteArrayOutputStream);
+		out.writeObject(testRecord);
+		out.flush();
+		return  byteArrayOutputStream.toByteArray();
+	}
+
+	private static MessageBuilder<byte[]> createBaseMessage(byte[] payload, String contentType) {
+		return MessageBuilder.withPayload(payload)
+				.setHeader(MessageHeaders.CONTENT_TYPE, contentType);
+	}
+
+	@Configuration
+	@EnableIntegration
+	public static class ContextConfiguration {
+
+		private static final ExpressionParser parser = new SpelExpressionParser();
+
+		@Bean
+		public ToCloudEventTransformer transformerWithNoExtensions() {
+			return new ToCloudEventTransformer();
+		}
+
+		@Bean
+		public ToCloudEventTransformer transformerWithExtensions() {
+			return new ToCloudEventTransformer(TEST_PATTERNS);
+		}
+
+		@Bean
+		public ToCloudEventTransformer transformerWithNoExtensionsNoFormat() {
+			ToCloudEventTransformer toCloudEventsTransformer =  new ToCloudEventTransformer();
+			toCloudEventsTransformer.setFailOnNoFormat(false);
+			return toCloudEventsTransformer;
+		}
+
+		@Bean
+		public ToCloudEventTransformer transformerWithExtensionsNoFormat() {
+			ToCloudEventTransformer toCloudEventsTransformer =  new ToCloudEventTransformer(TEST_PATTERNS);
+			toCloudEventsTransformer.setFailOnNoFormat(false);
+			return toCloudEventsTransformer;
+		}
+
+		@Bean
+		public ToCloudEventTransformer transformerWithExtensionsNoFormatWithPrefix() {
+			ToCloudEventTransformer toCloudEventsTransformer =  new ToCloudEventTransformer(TEST_PATTERNS);
+			toCloudEventsTransformer.setFailOnNoFormat(false);
+			toCloudEventsTransformer.setCloudEventPrefix("CLOUDEVENTS-");
+			return toCloudEventsTransformer;
+		}
+
+		@Bean
+		public ToCloudEventTransformer transformerWithInvalidIDExpression() {
+			ToCloudEventTransformer transformer = new ToCloudEventTransformer();
+			transformer.setEventIdExpression(parser.parseExpression("null"));
+			return transformer;
+		}
+
+		@Bean
+		public ToCloudEventTransformer transformerWithNoExtensionsNoFormatEnabled() {
+			ToCloudEventTransformer toCloudEventsTransformer =  new ToCloudEventTransformer();
```

Comment:
Please, do some reformatting with your IDE: all those extra spaces should go away.